src/main/input.js

```diff
@@ -30,6 +31,281 @@ export const axis = {
   "rA" : 17  // R button analog sensor
 };
 
+export function inputData ( list ) {
+  this.a   = list[button["a"  ]];
+  this.b   = list[button["b"  ]];
+  this.x   = list[button["x"  ]];
+  this.y   = list[button["y"  ]];
+  this.z   = list[button["z"  ]];
+  this.r   = list[button["r"  ]];
+  this.l   = list[button["l"  ]];
+  this.s   = list[button["s"  ]];
+  this.du  = list[button["du" ]]; 
+  this.dr  = list[button["dr" ]];
+  this.dd  = list[button["dd" ]];
+  this.dl  = list[button["dl" ]];
+  this.lsX = list[  axis["lsX"]];
+  this.lsY = list[  axis["lsY"]];
+  this.csX = list[  axis["csX"]];
+  this.csY = list[  axis["csY"]];
+  this.lA  = list[  axis["lA" ]];
+  this.rA  = list[  axis["rA" ]];
+  this.rawX = list[18];
+  this.rawY = list[19];
+};
+
+const nullInput = new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0]);
+
+export const nullInputs = [ new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          , new inputData ( [false, false, false, false, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0, 0, 0])
+                          ];
```

Comment:
These blocks are pretty silly. Can we just have the inputData constructor have a default arg if none is supplied? And can we have a constant for how long `nullInputs` is supposed to be, instead of having 8 be this magic number?